source/s3_auto_ranged_get.c

```diff
@@ -651,11 +699,16 @@ static int s_discover_object_range_and_content_length(
 
                 break;
             }
-
-            /* When discovering the object size via first-part, the object range is the entire object. */
-            object_range_start = 0;
-            object_range_end = total_content_length - 1; /* range-end is inclusive */
-
+            if (auto_ranged_get->initial_message_has_range_header) {
+                object_range_end = object_size - 1;
+                if (auto_ranged_get->initial_message_has_end_range) {
+                    object_range_end = aws_min_u64(object_range_end, auto_ranged_get->initial_object_range_end);
+                }
```

Comment:
utterly trivial: reads better as if/else
```suggestion
                if (auto_ranged_get->initial_message_has_end_range) {
                    object_range_end = aws_min_u64(object_range_end, auto_ranged_get->initial_object_range_end);
                } else {
                    object_range_end = object_size - 1;
                }
```

Reply:
The updated `object_range_end` is used in `aws_min_u64`. I have changed it to an if/else structure and updated that line to directly use `object_size - 1`.